libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteArrow.java

```diff
@@ -110,8 +110,10 @@
   void addUpcomingManeuverArrow(RouteProgress routeProgress) {
     boolean invalidUpcomingStepPoints = routeProgress.upcomingStepPoints() == null
             || routeProgress.upcomingStepPoints().size() < TWO_POINTS;
-    boolean invalidCurrentStepPoints =
-      routeProgress.currentLegProgress().currentStepProgress().stepPoints().size() < TWO_POINTS;
+    boolean invalidCurrentStepPoints = routeProgress.currentLegProgress() != null
+            && routeProgress.currentLegProgress().currentStepProgress() != null
+            && routeProgress.currentLegProgress().currentStepProgress().stepPoints() != null
```

Comment:
These logical checks should be switched the other way around.

Reply:
There's a NavigationPuckPresenter which updates the puck drawable.

Reply:
Regarding MapRouteArrow, what do you mean they should be switched around? If i put the third one first and the call to currentLegProgress() returns null then the next call to currentStepProgress() will cause a NPE.

Reply:
I'll remove the unused updateCurrentLocationDrawable method.